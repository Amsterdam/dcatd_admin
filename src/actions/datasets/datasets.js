import { fetchSchema } from '../schema/schema';
import api from '../../services/api/api';

export const FETCH_DATASETS_SUCCESS = 'FETCH_DATASETS_SUCCESS';
export const FETCH_DATASETS_FAILURE = 'FETCH_DATASETS_FAILURE';

export function fetchDatasetsSuccess(datasets) {
  return {
    type: FETCH_DATASETS_SUCCESS,
    datasets
  };
}

export function fetchDatasetsFailure(error) {
  return { type: FETCH_DATASETS_FAILURE, error };
}

export function fetchDatasets() {
  return (dispatch) => {
    return dispatch(fetchSchema()).then(() => {
      return fetch(api.datasets)
        .then((response) => response.json())
        .then((response) => response['dcat:dataset'].map((dataset) => ({
          id: dataset['dct:identifier'],
          title: dataset['dct:title'] || '',
          description: dataset['dct:description'] || ''
        })))
        .then((datasets) => dispatch(fetchDatasetsSuccess(datasets)));
    }).catch((error) => dispatch(fetchDatasetsFailure(error)));
  };
}
