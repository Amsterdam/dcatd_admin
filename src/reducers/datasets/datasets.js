import {
  FETCH_DATASETS_SUCCESS
} from '../../actions/datasets/datasets';
import {
  CREATE_DATASET_SUCCESS,
  UPDATE_DATASET_SUCCESS,
  REMOVE_DATASET_SUCCESS
} from '../../actions/dataset/dataset';

const initialState = [];

export default function (state = initialState, action) {
  switch (action.type) {
    case FETCH_DATASETS_SUCCESS:
      return [...action.datasets];

    case CREATE_DATASET_SUCCESS:
    case UPDATE_DATASET_SUCCESS:
    case REMOVE_DATASET_SUCCESS:
      return [];

    default:
      return state;
  }
}
