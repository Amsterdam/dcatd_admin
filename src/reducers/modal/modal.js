import { SET_MODAL_SUCCESS } from '../../actions/modal/modal';

const initialState = {
  actionLabel: 'OK',
  cancelLabel: 'Annuleren',
  content: 'Er gebeurt iets.',
  open: false,
  onProceed: () => {}
};

// eslint-disable-next-line default-param-last
export default function modal(state = initialState, action) {
  switch (action.type) {
    case SET_MODAL_SUCCESS:
      return {
        ...action.modal
      };

    default:
      return state;
  }
}
