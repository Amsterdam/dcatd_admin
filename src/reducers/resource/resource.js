import { SET_RESOURCE_SUCCESS, EMPTY_RESOURCE_SUCCESS } from '../../actions/resource/resource';

const initialState = {};

// eslint-disable-next-line default-param-last
export default function resource(state = initialState, action) {
  switch (action.type) {
    case SET_RESOURCE_SUCCESS:
      return {
        'dct:modified': new Date().toISOString().split('T')[0],
        ...action.resource
      };

    case EMPTY_RESOURCE_SUCCESS:
      return {};

    default:
      return state;
  }
}
