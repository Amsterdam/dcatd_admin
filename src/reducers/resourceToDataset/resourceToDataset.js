import { SET_RESOURCE_TO_DATASET_SUCCESS } from '../../actions/resourceToDataset/resourceToDataset';

const initialState = {};

// eslint-disable-next-line default-param-last
export default function resourceToDataset(state = initialState, action) {
  switch (action.type) {
    case SET_RESOURCE_TO_DATASET_SUCCESS:
      return {
        ...action.resource
      };

    default:
      return state;
  }
}
