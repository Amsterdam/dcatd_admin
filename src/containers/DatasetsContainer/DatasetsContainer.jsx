import React from 'react';
import PropTypes from 'prop-types';
import { Route } from 'react-router';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';

import { fetchDataset, emptyDataset, createDataset, cancelDataset, removeDataset, updateDataset }
  from '../../actions/dataset/dataset';
import { setResourceToDataset } from '../../actions/resourceToDataset/resourceToDataset';
import { emptyResource, setResource } from '../../actions/resource/resource';
import { setModal } from '../../actions/modal/modal';

import DatasetList from '../../components/DatasetList/DatasetList';
import DatasetForm from '../../containers/DatasetForm/DatasetForm';
import ResourceForm from '../../containers/ResourceForm/ResourceForm';

import './datasets-container.scss';
import { fetchDatasets } from '../../actions/datasets/datasets';

const mapStateToProps = state => ({
  dataset: state.dataset,
  datasets: state.datasets,
  resource: state.resource,
  resourceToDataset: state.resourceToDataset,
  schema: state.schema,
  uiDataset: state.uiDataset,
  uiResource: state.uiResource
});

const mapDispatchToProps = dispatch => bindActionCreators({
  onFetch: fetchDataset,
  onEmpty: emptyDataset,
  onCreate: createDataset,
  onCancel: cancelDataset,
  onRemove: removeDataset,
  onUpdate: updateDataset,
  onEmptyResource: emptyResource,
  onSetResource: setResource,
  onSetModal: setModal,
  onSetResourceToDataset: setResourceToDataset,
  onFetchDatasets: fetchDatasets
}, dispatch);

const DatasetsContainer = props => (
  <section>
    <Route
      exact
      path="/datasets"
      render={() => (
        <DatasetList
          datasets={props.datasets}
          onFetchDatasets={props.onFetchDatasets}
        />
      )}
    />
    <Route
      exact
      path="/datasets/:id([\w-]{2,})"
      render={() => (
        <div
          className={`form-wrapper
            form-wrapper--${props.resource['ams:resourceType'] ? 'show' : 'hide'}-resource-form`}
        >
          <ResourceForm
            schema={(props.schema && props.schema.properties &&
              props.schema.properties['dcat:distribution'] &&
              props.schema.properties['dcat:distribution'].items) || {}}
            uiResource={props.uiResource}
            formData={props.resource}
            onSetResourceToDataset={props.onSetResourceToDataset}
            onEmptyResource={props.onEmptyResource}
            setModal={props.onSetModal}
            datasetTitle={props.dataset['dct:title']}
          />
          <DatasetForm
            id={props.match.params.id}
            dataset={props.dataset}
            schema={props.schema}
            resourceToDataset={props.resourceToDataset}
            uiDataset={props.uiDataset}
            onFetch={props.onFetch}
            onEmpty={props.onEmpty}
            onUpdate={props.onUpdate}
            onCancel={props.onCancel}
            onRemove={props.onRemove}
            onEmptyResource={props.onEmptyResource}
            onSetResource={props.onSetResource}
            setModal={props.onSetModal}
          />
        </div>
      )}
    />
    <Route
      exact
      path="/datasets/_"
      render={() => (
        <div
          className={`form-wrapper
            form-wrapper--${props.resource['ams:resourceType'] ? 'show' : 'hide'}-resource-form`}
        >
          <ResourceForm
            schema={(props.schema && props.schema.properties &&
              props.schema.properties['dcat:distribution'] &&
              props.schema.properties['dcat:distribution'].items) || {}}
            uiResource={props.uiResource}
            formData={props.resource}
            onSetResourceToDataset={props.onSetResourceToDataset}
            onEmptyResource={props.onEmptyResource}
            setModal={props.onSetModal}
          />
          <DatasetForm
            schema={props.schema}
            resourceToDataset={props.resourceToDataset}
            uiDataset={props.uiDataset}
            onCreate={props.onCreate}
            onCancel={props.onCancel}
            onEmpty={props.onEmpty}
            onEmptyResource={props.onEmptyResource}
            onSetResource={props.onSetResource}
            setModal={props.onSetModal}
          />
        </div>
      )}
    />
  </section>
);

DatasetsContainer.defaultProps = {
  dataset: {},
  datasets: [],
  resource: {},
  schema: {},
  match: null,
  resourceToDataset: {},
  uiDataset: {},
  uiResource: {},

  onFetch: () => {}
};

DatasetsContainer.propTypes = {
  match: PropTypes.object,
  dataset: PropTypes.object,
  datasets: PropTypes.arrayOf(PropTypes.object),
  resource: PropTypes.object,
  resourceToDataset: PropTypes.object,

  onFetch: PropTypes.func.isRequired,
  onEmpty: PropTypes.func.isRequired,
  onCreate: PropTypes.func.isRequired,
  onRemove: PropTypes.func.isRequired,
  onUpdate: PropTypes.func.isRequired,
  onCancel: PropTypes.func.isRequired,
  onEmptyResource: PropTypes.func.isRequired,
  onSetResource: PropTypes.func.isRequired,
  onSetModal: PropTypes.func.isRequired,
  onSetResourceToDataset: PropTypes.func.isRequired,
  onFetchDatasets: PropTypes.func.isRequired,

  schema: PropTypes.object,
  uiDataset: PropTypes.object,
  uiResource: PropTypes.object
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(DatasetsContainer);
