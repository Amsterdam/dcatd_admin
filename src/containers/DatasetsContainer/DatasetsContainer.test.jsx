import React from 'react';
import configureMockStore from 'redux-mock-store';
import { shallow } from 'enzyme';
import thunk from 'redux-thunk';

import DatasetsContainer from './DatasetsContainer';

import { fetchDataset, emptyDataset, createDataset, cancelDataset, removeDataset, updateDataset }
  from '../../actions/dataset/dataset';
import { setResourceToDataset } from '../../actions/resourceToDataset/resourceToDataset';
import { emptyResource, setResource } from '../../actions/resource/resource';
import { setModal } from '../../actions/modal/modal';
import { fetchDatasets, FETCH_DATASETS_SUCCESS } from '../../actions/datasets/datasets';

jest.mock('../../actions/dataset/dataset');
jest.mock('../../actions/resourceToDataset/resourceToDataset');
jest.mock('../../actions/resource/resource');
jest.mock('../../actions/modal/modal');
jest.mock('../../actions/datasets/datasets');

const middlewares = [thunk];

describe('DatasetContainer', () => {
  let store;
  let wrap;

  beforeEach(() => {
    fetchDataset.mockImplementation(() => ({ type: 'FETCH_DATASET_SUCCESS' }));
    emptyDataset.mockImplementation(() => ({ type: 'EMPTY_DATASET_SUCCESS' }));
    createDataset.mockImplementation(() => ({ type: 'CREATE_DATASET_SUCCESS' }));
    cancelDataset.mockImplementation(() => ({ type: 'CANCEL_DATASET_SUCCESS' }));
    removeDataset.mockImplementation(() => ({ type: 'REMOVE_DATASET_SUCCESS' }));
    updateDataset.mockImplementation(() => ({ type: 'UPDATE_DATASET_SUCCESS' }));
    emptyResource.mockImplementation(() => ({ type: 'EMPTY_RESOURCE_SUCCESS' }));
    setResource.mockImplementation(() => ({ type: 'SET_RESOURCE_SUCCESS' }));
    setModal.mockImplementation(() => ({ type: 'SET_MODAL_SUCCESS' }));
    setResourceToDataset.mockImplementation(() => ({ type: 'SET_RESOURCE_TO_DATASET_SUCCESS' }));
    fetchDatasets.mockImplementation(() => ({ type: FETCH_DATASETS_SUCCESS }));
  });

  afterEach(() => {
    fetchDataset.mockReset();
    emptyDataset.mockReset();
    cancelDataset.mockReset();
    cancelDataset.mockReset();
    removeDataset.mockReset();
    updateDataset.mockReset();
    emptyResource.mockReset();
    setResource.mockReset();
    setModal.mockReset();
    setResourceToDataset.mockReset();
    fetchDatasets.mockReset();
  });

  describe('renders component', () => {
    it('renders default props', () => {
      store = configureMockStore(middlewares)();
      wrap = shallow(<DatasetsContainer />, { context: { store } }).dive();
      expect(wrap).toMatchSnapshot();
    });
  });
});
