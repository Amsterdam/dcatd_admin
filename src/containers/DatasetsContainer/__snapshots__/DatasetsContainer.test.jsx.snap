// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatasetContainer renders component renders default props 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DatasetsContainer
    dataset={Object {}}
    datasets={Array []}
    match={null}
    onCancel={[Function]}
    onCreate={[Function]}
    onEmpty={[Function]}
    onEmptyResource={[Function]}
    onFetch={[Function]}
    onFetchDatasets={[Function]}
    onRemove={[Function]}
    onSetModal={[Function]}
    onSetResource={[Function]}
    onSetResourceToDataset={[Function]}
    onUpdate={[Function]}
    resource={Object {}}
    resourceToDataset={Object {}}
    schema={Object {}}
    uiDataset={Object {}}
    uiResource={Object {}}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Route
          exact={true}
          path="/datasets"
          render={[Function]}
        />,
        <Route
          exact={true}
          path="/datasets/:id([\\\\w-]{2,})"
          render={[Function]}
        />,
        <Route
          exact={true}
          path="/datasets/_"
          render={[Function]}
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "exact": true,
          "path": "/datasets",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "exact": true,
          "path": "/datasets/:id([\\\\w-]{2,})",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "exact": true,
          "path": "/datasets/_",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Route
            exact={true}
            path="/datasets"
            render={[Function]}
          />,
          <Route
            exact={true}
            path="/datasets/:id([\\\\w-]{2,})"
            render={[Function]}
          />,
          <Route
            exact={true}
            path="/datasets/_"
            render={[Function]}
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/datasets",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/datasets/:id([\\\\w-]{2,})",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/datasets/_",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "context": Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
  },
}
`;
